sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -392,6 +400,47 @@ rom_error_t flash_ctrl_data_erase(uint32_t addr,
   return wait_for_done(kErrorFlashCtrlDataErase);
 }
 
+rom_error_t flash_ctrl_data_erase_verify(uint32_t addr,
+                                         flash_ctrl_erase_type_t erase_type) {
+  size_t byte_count;
+  switch (launder32(erase_type)) {
+    case kFlashCtrlEraseTypeBank:
+      HARDENED_CHECK_EQ(erase_type, kFlashCtrlEraseTypeBank);
+      byte_count = FLASH_CTRL_PARAM_BYTES_PER_BANK;
+      break;
+    case kFlashCtrlEraseTypePage:
+      HARDENED_CHECK_EQ(erase_type, kFlashCtrlEraseTypePage);
+      byte_count = FLASH_CTRL_PARAM_BYTES_PER_PAGE;
+      break;
+    default:
+      HARDENED_UNREACHABLE();
+  }
+
+  // Truncate to the closest lower bank/page aligned address.
+  static_assert(__builtin_popcount(FLASH_CTRL_PARAM_BYTES_PER_BANK) == 1,
+                "Bytes per bank must be a power of two.");
+  static_assert(__builtin_popcount(FLASH_CTRL_PARAM_BYTES_PER_PAGE) == 1,
+                "Bytes per page must be a power of two.");
+  addr &= (~byte_count + 1);
+  rom_error_t error = kErrorOk;
```

Comment:
optional: is it worth considering incorporating the byte_count into the error value?

```c
rom_error_t error = kErrorOk - byte_count;
...
  error += byte_count;
...
  return error;
```

Reply:
I think so. Added this with two small modifications: (1) `error` is initialized in the switch statement so that `kErrorOk` does not end up in a register, (2) probably not a big difference but xored with `byte_count - 1` instead of add/subtract to touch the lower bits.